pkg/nsxt/nsxt_secret.go

```diff
@@ -0,0 +1,152 @@
+/*
+ Copyright 2021 The Kubernetes Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package nsxt
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/core"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/security"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/client-go/informers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+// SecretManager is used to manage NSXT secret
+type SecretManager struct {
+	secretName      string
+	secretNamespace string
+	connectors      []*client.RestConnector
+}
+
+// secretManager is a singleton for all NSXT secret consumers
+var secretManager *SecretManager
```

Comment:
Yes, only `NewNsxtConnector` calls `NewSecretManager`, but Route provider and LB provider will both call `NewNsxtConnector`. 
Right now loading nsxt config is inside LB and Route provider, so we need a singleton to update connectors for both of them

Reply:
Maybe ideally NSXT config should be loaded and initialized in VSphere's `buildVSphereFromConfig` and `Initialize` separately, but for the time being I thought a singleton could work :)

Reply:
I'd like to avoid singletons in possible, especially if it's managed through a global state. I do think it's problematic that we are calling `NewNsxtConnector` twice, do you think we can refactor the code to call it once and pass a single instance of the connecter to both the LB and routes controllers?

Reply:
Calling `NewNsxtConnector` twice because I planed to use different connections with NSXT for Route and LB. But maybe it's better to share the same connector. 
Then in this case connector and secret_manager should be called once, let me try to move NSXT init logic in VSphere directly, and pass them to Route and LB provider.